File: Backend/hospital/hospital/src/main/java/com/bloodbank/hospital/services/patientService.java
package com.bloodbank.hospital.services;

import java.util.Optional;

import com.bloodbank.hospital.model.patient;

public interface patientService {
	public Iterable<patient> allPatients();
	public Optional<patient> getPatientById(int id);
	public Optional<patient> getByNameAndEmail(String name,String email);
	public patient addPatient(patient p);
	public int findIdByNameAndEmail(String name,String email);
}

File: blood bank front end/blood-bank/src/Services/PatientService.js
import axios from 'axios';



class PatientService{
getPatientIdByNameAndEmail(patient)
    {
        const Patient_Base_URL=`http://localhost:8080/hospital/patient-id/${patient.name}/${patient.email}`;
        return axios.get(Patient_Base_URL);
    }


    
getPatientById(id)
{
    const patient_by_id_url=`http://localhost:8080/hospital/patient-list/${id}`;
    return axios.get(patient_by_id_url);
}

sendSms(sms)
{
    const sms_url=`http://localhost:8080/sms`;
    return axios.post(sms_url,sms);
}

addPatient(patient)
    {
        const add_patient_url= `http://localhost:8080/hospital/addPatient`;
        return axios.post(add_patient_url,patient);
    }


}


export default new PatientService();